application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,268 @@
+        return guild.getChannels()
+            .stream()
+            .map(GuildChannel::getIdLong)
+            .filter(channelsToMonitorById::containsKey)
+            .map(channelsToMonitorById::get)
+            .toList();
+    }
+
+    /**
+     * Creates the status message (specific to the guild specified) that shows which channels are
+     * busy/free.
+     * <p>
+     * It first updates the channel names, order and grouping(categories) according to
+     * {@link net.dv8tion.jda.api.JDA} for the monitored channels. So that the output is always
+     * consistent with remote changes.
+     * 
+     * @param guild the guild the message is intended for.
+     * @return a string representing the busy/free status of channels in this guild. The String
+     *         includes emojis and other discord specific markup. Attempting to display this
+     *         somewhere other than discord will lead to unexpected results.
+     */
+    public String statusMessage(@NotNull final Guild guild) {
+        List<ChannelStatus> statusFor = guildMonitoredChannelsList(guild);
+
+        // update name so that current channel name is used
+        statusFor.forEach(channelStatus -> channelStatus.updateChannelName(guild));
+
+        // dynamically separate channels by channel categories
+        StringBuilder sb = new StringBuilder();
+        String categoryName = "";
+        for (ChannelStatus status : statusFor) {
+            Category category = guild.getGuildChannelById(status.getChannelId()).getParent();
+            if (category != null && !category.getName().equals(categoryName)) {
+                categoryName = category.getName();
+                // append the category name on a new line with markup for underlining
+                // FIXME possible bug when not all channels are part of categories, may mistakenly
+                // include uncategoried channels inside previous category. will an uncategoried
+                // channel return an empty string or null? javadocs dont say.
+                sb.append("\n__").append(categoryName).append("__\n");
+            }
+            sb.append(status.toDiscord()).append("\n");
+        }
+
+        return sb.toString();
+    }
+
+    /**
+     * This method checks all channels in a guild that is currently being monitored and are
+     * currently busy and determines if the last time it was updated is more than an hour ago. If so
+     * it changes the channel's status to free.
+     * <p>
+     * This method is run automatically during startup and should be run on a set schedule, as
+     * defined in {@link org.togetherjava.tjbot.config.FreeCommandConfig}. The scheduled execution
+     * is not currently implemented
+     * 
+     * @param guild the guild for which to test the channel statuses of.
+     */
+    public void updateStatusFor(@NotNull Guild guild) {
+        guildMonitoredChannelsList(guild).parallelStream()
```

Comment:
`parallelStream()` is only worth if the list is either really long (>10k) or your operations are heavy and blocking. Is that the case? doublecheck.

Reply:
its blocking
... it includes rest calls